model_specific/SM/standard_model.cpp

```diff
@@ -613,10 +614,19 @@ void Standard_model::calculate_pole_masses()
    tp.run_task([this] () { calculate_MFd_pole(); });
    tp.run_task([this] () { calculate_MFu_pole(); });
    tp.run_task([this] () { calculate_MFe_pole(); });
-   tp.run_task([this] () { calculate_MVWp_pole(); });
-
+   tp.run_task([this] () {
+      // if mW=0 it means that the SM was not initialized via matching to the SM
+      // compute mW using ordinary 1-loop calculation
+      if (!(PHYSICAL(MVWp) > 0.)) {
```

Comment:
I'm wondering if the test should be `if (PHYSICAL(MVWp) == 0.)`.
Reason: The current condition would be true if the mass is `<= 0`, i.e. a tachyonic mass would also trigger the re-calculation of the W boson mass.

Reply:
So maybe `if (!(std::abs(PHYSICAL(MVWp)) > 0.))`? I wrote it like that because I'm always afraid of comparing a double with 0.

Reply:
I know what you mean. However, if we have initialized `MVWp = 0`, then the statement `PHYSICAL(MVWp) == 0.` is true if `MVWp` was not touched (computed). So, this is safe. :)